modules/photo/src/ccm/utils.hpp

```diff
@@ -0,0 +1,153 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Author: Longbu Wang <wanglongbu@huawei.com.com>
+//         Jinheng Zhang <zhangjinheng1@huawei.com>
+//         Chenqi Shan <shanchenqi@huawei.com>
+
+#ifndef __OPENCV_CCM_UTILS_HPP__
+#define __OPENCV_CCM_UTILS_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+namespace ccm {
+
+double gammaCorrection_(const double& element, const double& gamma);
+
+/** @brief gamma correction.
+           \f[
+            C_l=C_n^{\gamma},\qquad C_n\ge0\\
+            C_l=-(-C_n)^{\gamma},\qquad C_n<0\\\\
+            \f]
+    @param src the input array,type of Mat.
+    @param gamma a constant for gamma correction.
+    @param dst the output array, type of Mat.
+ */
+Mat gammaCorrection(const Mat& src, const double& gamma, Mat dst=Mat());
+
+/** @brief maskCopyTo a function to delete unsatisfied elementwise.
+    @param src the input array, type of Mat.
+    @param mask operation mask that used to choose satisfided elementwise.
+ */
+Mat maskCopyTo(const Mat& src, const Mat& mask);
+
+/** @brief multiple the function used to compute an array with n channels
+      mulipied by ccm.
+    @param xyz the input array, type of Mat.
+    @param ccm the ccm matrix to make color correction.
+ */
+Mat multiple(const Mat& xyz, const Mat& ccm);
```

Comment:
Most probably the function may be replaced with cv::transform https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga393164aa54bb9169ce0a8cc44e08ff22

Reply:
Still relevant.